### PR TITLE
docker: adding ubuntu20.04

### PR DESCRIPTION
**Pull request by RikkiButler20**

### CODE REVIEW

File: share/spack/docker/ubuntu-2010.dockerfile [hunk 1]
```diff
@@ -0,0 +1,76 @@
+FROM ubuntu:20.10
```
Comment (alalazo) [inline share/spack/docker/ubuntu-2010.dockerfile]: Having a recent Ubuntu is great, thanks! Wondering if we should:
```suggestion
FROM ubuntu:20.04
```
to have all our Spack + Ubuntu images based on LTS.

File: share/spack/docker/ubuntu-2004.dockerfile [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        /root/.spack/modules.yaml \
+ && rm -rf /root/*.* /run/nologin $SPACK_ROOT/.git
+
+RUN [ -f ~/.profile ]                                               \
+ && sed -i 's/mesg n/( tty -s \&\& mesg n || true )/g' ~/.profile \
+ || true
+
+# [WORKAROUND]
+# https://bugs.launchpad.net/ubuntu/+source/lua-posix/+bug/1752082
+RUN ln -s posix_c.so /usr/lib/x86_64-linux-gnu/lua/5.2/posix.so
```
Comment (alalazo) [inline share/spack/docker/ubuntu-2004.dockerfile:67]: Are you able to build this image? For me it fails with:
```
Step 17/22 : RUN ln -s posix_c.so /usr/lib/x86_64-linux-gnu/lua/5.2/posix.so
 ---> Running in 1fe0af8be2ab
ln: failed to create symbolic link '/usr/lib/x86_64-linux-gnu/lua/5.2/posix.so': File exists
The command '/bin/sh -c ln -s posix_c.so /usr/lib/x86_64-linux-gnu/lua/5.2/posix.so' returned a non-zero code: 1
```
Maybe the bug was specific to Ubuntu 18.04.

**Review by alalazo — CHANGES_REQUESTED** [review]
Can you check if the image builds for you?

**Comment by vsoch** [comment]
Any updates here? I'd like this added.

**Comment by tldahlgren** [comment]
@RikkiButler20 Please resolve conflicts.  There are also some questions here.

**Pull request closed without merging**